test/NftReward.t.sol

```diff
@@ -47,7 +57,8 @@ contract NftRewardTest is Test {
         // get mint request digest which should be signed
         bytes32 digest = nftReward.getMintRequestDigest(mintRequest);
 
-        assertEq(digest, 0x2c680706f2350ed5622f229af6736cd20626f7b9b4569b2fd5abb7e086886dc3);
+        // expected digest: 0x2c680706f2350ed5622f229af6736cd20626f7b9b4569b2fd5abb7e086886dc3
+        assertEq(digest, 0xf85127466b4ea4a97ec0c1b445b54622c8c3760b720bf9136b491d50f122b043);
```

Comment:
@gpylypchuk I don't understand why was the hash updated if the signed data is untouched?

Reply:
Answering my own question, hash has changed because sender's nonce is used on smart contract address computation which is a part of EIP721. This PR adds more transactions in the test file, hence some other nonce is used, hence the hash is different.